server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -207,6 +210,15 @@ private RolloverResult rolloverAlias(ClusterState currentState, IndexAbstraction
     private RolloverResult rolloverDataStream(ClusterState currentState, IndexAbstraction.DataStream dataStream, String dataStreamName,
                                               CreateIndexRequest createIndexRequest, List<Condition<?>> metConditions,
                                               boolean silent, boolean onlyValidate) throws Exception {
+
+        if (SnapshotsService.snapshottingDataStreams(currentState, Collections.singleton(dataStream.getName())).isEmpty() == false) {
```

Comment:
Can we add a comment explaining why we cannot rollover the datastream if a non-partial snapshot is already running?

Reply:
++ added one :)